concrete/src/Page/Collection/Version/Version.php

```diff
@@ -322,12 +326,47 @@ public function getAttributeValueObject($ak, $createIfNotExists = false)
      * Is this version approved?
      *
      * @return bool|int|string
+     *
+     * @see isApprovedNow()
      */
     public function isApproved()
     {
         return $this->cvIsApproved;
     }
 
+    /**
+     * Is this version approved and in the publish interval?
+     *
+     * @var string|int|\DateTime|null $when a date/time representation (empty: now)
+     * @return bool
+     */
+    public function isApprovedNow($when = null)
+    {
+        if (!$this->isApproved()) {
+            return false;
+        }
+        $start = $this->getPublishDate();
+        $end = $this->getPublishEndDate();
+        if (!$start && !$end) {
+            return true;
+        }
+        $dh = Facade::getFacadeApplication()->make('date');
+        if ($when) {
+            $when = $dh->toDB($when);
+        }
+        if (!$when) {
+            $when = $dh->getOverridableNow();
+        }
+        if ($start && $start > $now) {
+            return false;
+        }
+        if ($end && $end < $now) {
```

Comment:
`$now` should be `$when`? As the former isn't declared and later receives a value that isn't used?

Reply:
You are right: see https://github.com/concrete5/concrete5/pull/7510